cogs/moderation.py

```diff
@@ -51,11 +79,72 @@ async def ban(self, ctx: Context, member: discord.Member, *, reason: str = "No R
         await member.send(embed=embed)
         await member.ban(reason=reason)
 
+    class ActionReason(Converter):
+        async def convert(self, ctx: Context, argument: str) -> str:
+            ret = f"{ctx.author} (ID: {ctx.author.id}): {argument}"
+
+            if len(ret) > 512:
+                reason_max = 512 - len(ret) + len(argument)
+                raise BadArgument(f"Reason is too long ({len(argument)}/{reason_max})")
+            return ret
+
+    class MemberID(Converter):
```

Comment:
As you'll now be using the new `resolve_user` instead of `resolve_member` (as suggested) you might want to rename this to `UserID` instead